### PR TITLE
[Utilities] Updated utility references to new naming scheme

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Updated utility references to new naming scheme

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Update to documentation

### CODE REVIEW

File: utilities/pipelines/resourceDeployment/New-TemplateDeployment.ps1 [hunk 1]
```diff
@@ -362,7 +362,7 @@ Optional. ID of the subscription to deploy into. Mandatory if deploying into a s
 Optional. Name of the management group to deploy into. Mandatory if deploying into a management group (management group level)
 
 .PARAMETER additionalTags
-Optional. Provde a Key Value Pair (Object) that will be appended to the Parameter file tags. Example: @{myKey = 'myValue',myKey2 = 'myValue2'}.
+Optional. Provde a Key Value Pair (Object) that will be appended to the Parameter file tags. Example: @{myKey = 'myValue', myKey2 = 'myValue2'}.
```
Comment (eriqua) [inline utilities/pipelines/resourceDeployment/New-TemplateDeployment.ps1:365]:
```suggestion
Optional. Provide a Key Value Pair (Object) that will be appended to the Parameter file tags. Example: @{myKey = 'myValue', myKey2 = 'myValue2'}.
```

File: utilities/pipelines/resourcePublish/Get-ModulesToPublish.ps1 [hunk 2]
```diff
@@ -179,15 +179,15 @@ Mandatory. Path to a main.bicep/json file.
 Optional. If true, the function will recurse up the folder structure to find the closest main.bicep/json file.
 
 .EXAMPLE
-Get-ParentModuleTemplateFile -TemplateFilePath 'C:\Repos\Azure\ResourceModules\modules\storage\storage-accounts\table-services\tables\main.bicep' -Recurse
+Get-ParentModuleTemplateFile -TemplateFilePath 'C:\Repos\Azure\ResourceModules\modules\storage\storage-account\table-service\table\main.bicep' -Recurse
 
-    Directory: C:\Repos\Azure\ResourceModules\modules\storage\storage-accounts\tableServices
+    Directory: C:\Repos\Azure\ResourceModules\modules\storage\storage-account\tableService
```
Comment (eriqua) [inline utilities/pipelines/resourcePublish/Get-ModulesToPublish.ps1:184]:
```suggestion
    Directory: C:\Repos\Azure\ResourceModules\modules\storage\storage-account\table-service
```

File: utilities/pipelines/resourcePublish/Get-TemplateSpecsName.ps1 [hunk 3]
```diff
@@ -26,8 +26,8 @@ function Get-TemplateSpecsName {
     $templateSpecIdentifier = $templateSpecIdentifier -replace 'microsoft', 'ms'
 
     # Shorten the name
-    # This is requried as certain modules generate names such as `MS.RecoveryServices.vaults.replicationFabrics.replicationProtectionContainers.replicationProtectionContainerMappings` which are longer than the allowed 90 characters for template specs
-    # Using the logic below, the name is shortened to `MS.RecoveryServices.vaults.replicationFabrics.replicationProtectionContainers.Mappings` which has 'only' 86 characters
+    # This is requried as certain modules generate names such as `ms.recovery-services.vault.replication-fabric.replication-protection-container.replication-protection-container-mapping` which are longer than the allowed 90 characters for template specs
```
Comment (eriqua) [inline utilities/pipelines/resourcePublish/Get-TemplateSpecsName.ps1:29]:
```suggestion
    # This is required as certain modules generate names such as `ms.recovery-services.vault.replication-fabric.replication-protection-container.replication-protection-container-mapping` which are longer than the allowed 90 characters for template specs
```

File: utilities/pipelines/resourcePublish/Get-TemplateSpecsName.ps1 [hunk 4]
```diff
@@ -26,8 +26,8 @@ function Get-TemplateSpecsName {
...
+    # Using the logic below, the name is shortened to `ms.recovery-services.vault.replication-fabric.replication-protection-container.mapping` which has 'only' 86 characters
```
Comment (eriqua) [inline utilities/pipelines/resourcePublish/Get-TemplateSpecsName.ps1:30]: Please check if still 86 chars

File: utilities/pipelines/resourcePublish/Get-TemplateSpecsName.ps1 [hunk 5]
```diff
@@ -36,7 +36,7 @@ function Get-TemplateSpecsName {
             $stringToCheck = $nameElems[($index + 1)]
 
             # If a name is replicated in a path, it is usually plural in the parent, and singular in the child path.
-            # For example: /virtualNetworks/ (plural) & /virtualNetworks/virtualNetworkPeerings/ (singular)
+            # For example: /virtual-networks/ (plural) & /virtual-network/virtualNetwork-peering/ (singular)
```
Comment (eriqua) [inline utilities/pipelines/resourcePublish/Get-TemplateSpecsName.ps1:39]:
```suggestion
            # For example: /virtual-networks/ (plural) & /virtual-network/virtual-network-peering/ (singular)
```

File: utilities/pipelines/resourcePublish/Get-TemplateSpecsName.ps1 [hunk 6]
```diff
@@ -36,7 +36,7 @@ function Get-TemplateSpecsName {
             $stringToCheck = $nameElems[($index + 1)]
 
             # If a name is replicated in a path, it is usually plural in the parent, and singular in the child path.
```
Comment (eriqua) [inline utilities/pipelines/resourcePublish/Get-TemplateSpecsName.ps1:38]: Is this still valid now that all chunks are singular?

File: utilities/tools/Update-RoleAssignmentList.ps1 [hunk 7]
```diff
@@ -163,11 +163,11 @@ function Update-RoleAssignmentList {
...
                 }
             }
-            # also updating the roles in the [Microsoft.Authorization/RoleAssignments] module,
+            # also updating the roles in the [authorization/role-assignments] module,
             # which needs to be triggered separately, as the roles are not stored in the nested_roleAssignments.bicep
             # and therefore it's not detected by the search
-            if ($PSCmdlet.ShouldProcess('Role Assignments for module [Microsoft.Authorization/RoleAssignments]', 'Update')) {
-                $null = Update-RoleAssignmentListInner -ProviderNamespace 'Microsoft.Authorization' -ResourceType 'RoleAssignments' -Verbose
+            if ($PSCmdlet.ShouldProcess('Role Assignments for module [authorization/role-assignments]', 'Update')) {
+                $null = Update-RoleAssignmentListInner -ProviderNamespace 'authorization' -ResourceType 'role-assignments' -Verbose
```
Comment (eriqua) [inline utilities/tools/Update-RoleAssignmentList.ps1:170]:
```suggestion
                $null = Update-RoleAssignmentListInner -ProviderNamespace 'authorization' -ResourceType 'role-assignment' -Verbose
```

File: utilities/tools/Update-RoleAssignmentList.ps1 [hunk 8]
```diff
@@ -163,11 +163,11 @@ function Update-RoleAssignmentList {
                     $null = Update-RoleAssignmentListInner -ProviderNamespace $provider -ResourceType $type -Verbose
                 }
             }
-            # also updating the roles in the [Microsoft.Authorization/RoleAssignments] module,
+            # also updating the roles in the [authorization/role-assignments] module,
             # which needs to be triggered separately, as the roles are not stored in the nested_roleAssignments.bicep
             # and therefore it's not detected by the search
-            if ($PSCmdlet.ShouldProcess('Role Assignments for module [Microsoft.Authorization/RoleAssignments]', 'Update')) {
-                $null = Update-RoleAssignmentListInner -ProviderNamespace 'Microsoft.Authorization' -ResourceType 'RoleAssignments' -Verbose
+            if ($PSCmdlet.ShouldProcess('Role Assignments for module [authorization/role-assignments]', 'Update')) {
```
Comment (eriqua) [inline utilities/tools/Update-RoleAssignmentList.ps1:169]:
```suggestion
            if ($PSCmdlet.ShouldProcess('Role Assignments for module [authorization/role-assignment]', 'Update')) {
```

File: utilities/pipelines/resourceRemoval/Initialize-DeploymentRemoval.ps1 [hunk 9]
```diff
@@ -18,9 +18,9 @@ Optional. The name of the resource group the deployment was happening in. Releva
 Optional. The ID of the management group to fetch deployments from. Relevant for management-group level deployments.
 
 .EXAMPLE
-Initialize-DeploymentRemoval -DeploymentName 'virtualWans-20211204T1812029146Z' -TemplateFilePath "$home/ResourceModules/modules/network/virtual-wans/main.bicep" -resourceGroupName 'test-virtualWan-rg'
+Initialize-DeploymentRemoval -DeploymentName 'virtualWans-20211204T1812029146Z' -TemplateFilePath "$home/ResourceModules/modules/network/virtual-wan/main.bicep" -resourceGroupName 'test-virtualWan-rg'
```
Comment (eriqua) [inline utilities/pipelines/resourceRemoval/Initialize-DeploymentRemoval.ps1:21]: Are deployment names still in the form of 'camelCase-date'?

File: utilities/pipelines/resourcePublish/Get-TemplateSpecsName.ps1 (hunk 4 above)
Comment (AlexanderSehr) [reply utilities/pipelines/resourcePublish/Get-TemplateSpecsName.ps1:30]: It does - suprisingly :D

File: utilities/pipelines/resourceRemoval/Initialize-DeploymentRemoval.ps1 (hunk 9 above)
Comment (AlexanderSehr) [reply utilities/pipelines/resourceRemoval/Initialize-DeploymentRemoval.ps1:21]: Good catch. They will orientate themselves based on the path. This code actually is quite outdated now

```PowerShell
if ($templateFilePath -match '.*(\\|\/)Microsoft.+') {
    # If we can assume we're operating in a module structure, we can further fetch the provider namespace & resource type
    $shortPathElem = (($templateFilePath -split 'Microsoft\.')[1] -replace '\\', '/') -split '/' # e.g., AppConfiguration, configurationStores, .test, common, main.test.bicep
    $providerNamespace = $shortPathElem[0] # e.g., AppConfiguration
    $providerNamespaceShort = ($providerNamespace -creplace '[^A-Z]').ToLower() # e.g., ac

    $resourceType = $shortPathElem[1] # e.g., configurationStores
    $resourceTypeShort = ('{0}{1}' -f ($resourceType.ToLower())[0], ($resourceType -creplace '[^A-Z]')).ToLower() # e.g. cs

    $testFolderShort = Split-Path (Split-Path $templateFilePath -Parent) -Leaf  # e.g., common

    $deploymentNamePrefix = "$providerNamespaceShort-$resourceTypeShort-$testFolderShort" # e.g., ac-cs-common
}
```

will update it

Comment (AlexanderSehr) [reply utilities/pipelines/resourceRemoval/Initialize-DeploymentRemoval.ps1]: Fixed the name generation & updated the example. It should be `n-vw-t1-20211204T1812029146Z`

**Pull request merged**